src/compiler/checker.ts

```diff
@@ -1511,8 +1515,8 @@ namespace ts {
                         isInExternalModule = true;
                         // falls through
                     case SyntaxKind.ModuleDeclaration:
-                        const moduleExports = getSymbolOfNode(location as SourceFile | ModuleDeclaration).exports!;
-                        if (location.kind === SyntaxKind.SourceFile || isAmbientModule(location)) {
+                        const moduleExports = getSymbolOfNode(location as SourceFile | ModuleDeclaration).exports || emptySymbols;
+                        if (location.kind === SyntaxKind.SourceFile || (isModuleDeclaration(location) && location.flags & NodeFlags.Ambient && !isGlobalScopeAugmentation(location))) {

```

Comment:
I assume this change is because of the better alias-following in the binder. If so, then shouldn't the definition of `isAmbientModule` change? And why exclude global scope augmentations now, when they're explicitly included in `isAmbientModule`?

Reply:
`isAmbientModule` just wasn't the right predicate here - we wanted to do this behavior for every ambient namespace-like construct, as ambient namespaces have the same implicit-export rules (when you go do a lookup) as ambient module declarations do. Also global augmentations don't, so needed to be explicitly filtered out~